packages/starlight/utils/navigation.ts

```diff
@@ -348,9 +329,25 @@ function sidebarFromDir(
 	);
 }
 
+/**
+ * Intermediate sidebar represents sidebar entries generated from the user config for a specific
+ * locale and do not contain any information about the current page.
+ * These representations are cached per locale to avoid regenerating them for each page.
+ * When generating the final sidebar for a page, the intermediate sidebar is cloned and the current
+ * page is marked as such.
+ *
+ * @see getSidebarFromIntermediateSidebar
+ */
+const intermediateSidebars = new Map<string | undefined, SidebarEntry[]>();
```

Comment:
Put a request out to @Fryuni [on Discord](https://discord.com/channels/830184174198718474/1070481941863878697/1303688158445309982) for a second opinion on this vs an LRU cache etc.

Reply:
Concluded it’s not an issue 🎉